src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -450,6 +469,32 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return cell;
 	}
 
+	/** Adds Paramters cell based on Notebook URI parameters */
+	private addParametersCell(notebookUriParams: string, hasParameterCell: boolean, parameterCellIndex: number, hasInjectedCell: boolean): void {
+		let uriParamsIndex = parameterCellIndex;
+		// Set new uri parameters as a Injected Parameters cell after original parameter cell
+		if (hasParameterCell) {
+			uriParamsIndex = parameterCellIndex + 1;
+			// Set the uri parameters after the injected parameter cell
+			if (hasInjectedCell) {
+				uriParamsIndex = uriParamsIndex + 1;
+			}
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isInjectedParameter = true;
+			this.cells[uriParamsIndex].source = [injectedParametersMsg].concat(this.cells[uriParamsIndex].source);
+		} else {
+			// Set new parameters as the parameters cell as the first cell in the notebook
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isParameter = true;
+		}
+	}
+
+	/** Add new notebook URI parameters cell to notebook */
+	private addUriParameterCell(uriParamsIndex, notebookUriParams): void {
```

Comment:
Missing types for method parameters.

Reply:
Having both addParametersCell and addUriParametersCell methods seems a bit confusing imo since their method descriptions are the same. Maybe it'd be better to remove this helper method and rename the top one to addUriParameterCell since that is more descriptive? Then injectedParametersMsg and notebookUriParams can be set as the source at the same time (line 484).